File: promedio/solutions/correct/nlehmann.cpp
#include <algorithm>
#include <cstdio>
#include <map>
#include <vector>
using namespace std;

int main() {
  int N, T;
  scanf("%d%d", &N, &T);

  vector<int> arr(N+1);
  for (int i = 1; i <= N; ++i) {
    scanf("%d", &arr[i]);
    arr[i] -= T;
    arr[i] += arr[i-1];
  }

  map<int, int> m;
  for (int i = 1; i <= N; ++i) {
    m[arr[i]] = i;
  }

  int maxi = 0;
  int maxl = 0;
  for (int i = 0; i <= N; ++i) {
    int len = m[arr[i]] - i;
    if (m[arr[i]] >= i && len > maxl) {
      maxi = i;
      maxl = len;
    }
  }

  if (maxl > 0 ) {
    printf("%d %d\n", maxi, maxl);
  } else {
    printf("-1\n");
  }

  return 0;
}

File: promedio/solutions/correct/nlehmann.java
import java.io.*;
import java.util.*;

class promedio {
    static public void main(String[] args) throws Exception {
        BufferedReader in = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st;

        st = new StringTokenizer(in.readLine());
        int N = Integer.parseInt(st.nextToken());
        int T = Integer.parseInt(st.nextToken());

        int arr[] = new int[N+1];
        st = new StringTokenizer(in.readLine());
        for (int i = 1; i <= N; i++) {
            arr[i] = Integer.parseInt(st.nextToken()) - T;
            arr[i] += arr[i-1];
        }

        Map<Integer, Integer> m = new TreeMap<Integer, Integer>();
        for (int i = 1; i <=N; ++i) {
            m.put(arr[i], i);
        }

        int maxi = 0;
        int maxl = 0;
        for (int i = 0; i <=N; ++i) {
            int v = arr[i];
            if (m.containsKey(v) && m.get(v) != i && m.get(v) - i > maxl) {
                maxi = i;
                maxl = m.get(v) - i;
            }
        }

        if (maxl > 0) {
            System.out.println(maxi + " " + maxl);
        } else {
            System.out.println("-1");
        }
    }
}
